### DISCUSSION TITLE
Multi-vendor marketplace inventory management

### DISCUSSION THREAD

**Discussion by Klorrd** [medusajs/medusa#5006] [Q&A] [answered]
We are trying to create a multi vendor marketplace using Medusa JS. We need to have multiple stock locations for every vendor. In this setup, do we need unique sales channels for every vendor account? How should we approach to accomplish above scenario?

**Comment by srindom** [comment, marked as answer]
This is pretty straightforward. You will need to do the following:
1. [Install @medusajs/inventory and @medusajs/stock-locations](https://docs.medusajs.com/modules/multiwarehouse/install-modules)
2. [Create Stock Locations for each Vendor](https://docs.medusajs.com/modules/multiwarehouse/admin/manage-stock-locations#create-a-stock-location)
3. [Create Sales Channels for each vendor](https://docs.medusajs.com/modules/sales-channels/admin/manage)
4. [Associate Stock Locations with Sales Channels based on the vendor.](https://docs.medusajs.com/modules/multiwarehouse/admin/manage-stock-locations#associate-a-stock-location-with-a-sales-channel)

**Reply by Klorrd** [reply]
Thank you very much for the quick and concise response. :pray:

Just want to verify this before proceeding. So basically we need unique sales channels for every vendor right? 
I was under the impression that inventories are associated with store and we have one sales channel which is our eCommerce site.
It appears to be that to be able to utilize inventory modules, we need unique sales channels for every vendor. 

Please correct me if I have got this wrong. Thanks again.

**Comment by mk0y** [comment]
I still don't understand what is "a vendor" here. Stock location and sales channel for each vendor, but where is a vendor as an entity in the medusa ecosystem? We also need multiple sellers, which is same as vendor here.

**Comment by mapmychannel** [comment]
Multi-vendor marketplace is a ecommerce platform that allow the ecommerce sellers all across the globe to sell different type of product in a single place. Every vendors must have to charge listing fee, platform fee, commissions and membership fee.

If you want to know more features of multi-vendor marketplaces then you can go with the below-mentioned link.

[Multichannel Inventory Management](https://www.mapmychannel.com/)
